core_backend/tests/api/test_dashboard.py

```diff
@@ -0,0 +1,204 @@
+from datetime import datetime
+from typing import AsyncGenerator, Tuple
+
+import pytest
+from dateutil.relativedelta import relativedelta
+from sqlalchemy import delete
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from core_backend.app.dashboard.models import (
+    get_content_feedback_stats,
+    get_query_count_stats,
+    get_response_feedback_stats,
+    get_urgency_stats,
+)
+from core_backend.app.question_answer.models import (
+    ContentFeedbackDB,
+    QueryDB,
+    ResponseFeedbackDB,
+    save_content_feedback_to_db,
+    save_response_feedback_to_db,
+    save_user_query_to_db,
+)
+from core_backend.app.question_answer.schemas import (
+    ContentFeedback,
+    QueryBase,
+    ResponseFeedbackBase,
+)
+from core_backend.app.schemas import FeedbackSentiment
+from core_backend.app.urgency_detection.models import (
+    UrgencyQueryDB,
+    UrgencyResponseDB,
+    save_urgency_query_to_db,
+    save_urgency_response_to_db,
+)
+from core_backend.app.urgency_detection.schemas import UrgencyQuery, UrgencyResponse
+
+
+class TestUrgencyDetectionStats:
+
+    @pytest.fixture(scope="function", params=[(0, 0), (0, 1), (1, 0), (3, 5)])
+    async def urgency_detection(
+        self,
+        request: pytest.FixtureRequest,
+        asession: AsyncSession,
+        user: pytest.FixtureRequest,
+    ) -> AsyncGenerator[Tuple[int, int], None]:
+        n_urgent, n_not_urgent = request.param
+        data = [(f"Test urgent query {i}", True) for i in range(n_urgent)]
+        data += [(f"Test not urgent query {i}", False) for i in range(n_not_urgent)]
+
+        urgency_query_ids = []
+        urgency_response_ids = []
+        for message_text, is_urgent in data:
+            urgency_query = UrgencyQuery(message_text=message_text)
+            urgency_query_db = await save_urgency_query_to_db(
+                1, "test_secret_key", urgency_query, asession
+            )
+
+            urgency_response = UrgencyResponse(
+                is_urgent=is_urgent, failed_rules=[], details={}
+            )
+            await save_urgency_response_to_db(
+                urgency_query_db, urgency_response, asession
+            )
+
+            urgency_query_ids.append(urgency_query_db.urgency_query_id)
+            urgency_response_ids.append(urgency_query_db.urgency_query_id)
+
+        yield (n_urgent, n_not_urgent)
```

Comment:
I think we should not yield `n_not_urgent` if we are not using it in the code logic.

Reply:
Ah, I am doing this for completeness. I'll update the test to just map this to `_`